smarty/sampler.py

```diff
@@ -97,6 +97,21 @@ def __init__(self, molecules, basetypes_filename, initialtypes_filename, decorat
         # Save bond list to use throughout
         bondset = [("-","singly"), ("=", "doubly"), ("#","triply"), (":", "aromatic")]
 
+        used_basetypes = list()
+        self.atomtypes_with_no_matches = set()
+        # Check all SMART strings that are used as a base type
+        for (smarts, atom_type) in self.basetypes:
+            check_flag = self.smarts_matches(smarts)
```

Comment:
This is a little knit picky, but I'm not sure I understand why this variable is called `check_flag`? It doesn't really matter what we call it, but it would be better if it was clear from the name what it was defining.